.ci/functional-tests-utils.sh

```diff
@@ -0,0 +1,377 @@
+  docker version
+}
+
+function installMvn() {
+  yum install --assumeyes -d1 centos-release-scl
+  yum install --assumeyes -d1 rh-maven33
+}
+
+function installNodejs() {
+  curl -sL https://rpm.nodesource.com/setup_10.x | bash -
+  yum install -y nodejs
+}
+
+function installYarn() {
+  yum-config-manager --add-repo https://dl.yarnpkg.com/rpm/yarn.repo
+  yum install -y yarn
+}
+
+function installGit() {
+  yum install --assumeyes -d1 git
+}
+
+function installWget() {
+  yum -y install wget
+}
+
+function installGssCompiler() {
+  yum install -y gcc-c++ make
+}
+
+function installDependencies() {
+  echo "======== Installing dependencies: ========"
+  start=$(date +%s)
+
+  installEpelRelease
+  installYQ
+  installStartDocker
+  installJQ
+  installOC
+  installGit
+  installWget
+  # Getting dependencies ready
+  yum install --assumeyes -d1 \
+    patch \
+    pcp \
+    bzip2 \
+    golang \
+    make \
+    java-1.8.0-openjdk \
+    java-1.8.0-openjdk-devel \
+    google-chrome
+  installMvn
+  installNodejs
+  installYarn
+
+  stop=$(date +%s)
+  install_dep_duration=$((stop - start))
+  echo "======== Installing all dependencies lasted $install_dep_duration seconds. ========"
+
+}
+
+function installKVM() {
+  echo "======== Start to install KVM virtual machine ========"
+
+  yum install -y qemu-kvm libvirt libvirt-python libguestfs-tools virt-install
+
+  curl -L https://github.com/dhiltgen/docker-machine-kvm/releases/download/v0.10.0/docker-machine-driver-kvm-centos7 -o /usr/local/bin/docker-machine-driver-kvm
+  chmod +x /usr/local/bin/docker-machine-driver-kvm
+
+  systemctl enable libvirtd
+  systemctl start libvirtd
+
+  virsh net-list --all
+  echo "======== KVM has been installed successfully ========"
+}
+
+function installAndStartMinishift() {
+  echo "======== Start to install minishift ========"
+  curl -Lo minishift.tgz https://github.com/minishift/minishift/releases/download/v1.34.2/minishift-1.34.2-linux-amd64.tgz
+  tar -xvf minishift.tgz --strip-components=1
+  chmod +x ./minishift
+  mv ./minishift /usr/local/bin/minishift
+
+  #Setup GitHub token for minishift
+  if [ -z "$CHE_BOT_GITHUB_TOKEN" ]
+  then
+    echo "\$CHE_BOT_GITHUB_TOKEN is empty. Minishift start might fail with GitGub API rate limit reached."
+  else
+    echo "\$CHE_BOT_GITHUB_TOKEN is set, checking limits."
+    GITHUB_RATE_REMAINING=$(curl -slL "https://api.github.com/rate_limit?access_token=$CHE_BOT_GITHUB_TOKEN" | jq .rate.remaining)
+    if [ "$GITHUB_RATE_REMAINING" -gt 1000 ]
+    then
+      echo "Github rate greater than 1000. Using che-bot token for minishift startup."
+      export MINISHIFT_GITHUB_API_TOKEN=$CHE_BOT_GITHUB_TOKEN
+    else
+      echo "Github rate is lower than 1000. *Not* using che-bot for minishift startup."
+      echo "If minishift startup fails, please try again later."
+    fi
+  fi
+
+  minishift version
+  minishift config set memory 14GB
+  minishift config set cpus 4
+
+  echo "======== Lunch minishift ========"
```

Comment:
*Launch*

:)

Reply:
Yeah that was a copy from the upstream script haha